x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/fielddata/LatLonGeometryRelationVisitorTests.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.spatial.index.fielddata;
+
+import org.apache.lucene.document.ShapeField;
+import org.apache.lucene.geo.Component2D;
+import org.apache.lucene.geo.LatLonGeometry;
+import org.apache.lucene.tests.geo.GeoTestUtil;
+import org.elasticsearch.geo.GeometryTestUtils;
+import org.elasticsearch.geometry.Geometry;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.spatial.util.GeoTestUtils;
+
+import java.util.function.Supplier;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class LatLonGeometryRelationVisitorTests extends ESTestCase {
+
+    public void testPoint() throws Exception {
+        doTestShapes(GeoTestUtil::nextPoint);
+    }
+
+    public void testLine() throws Exception {
+        doTestShapes(GeoTestUtil::nextLine);
+    }
+
+    public void testPolygon() throws Exception {
+        doTestShapes(GeoTestUtil::nextPolygon);
+    }
+
+    private void doTestShapes(Supplier<LatLonGeometry> supplier) throws Exception {
+        Geometry geometry = GeometryTestUtils.randomGeometryWithoutCircle(0, false);
+        GeoShapeValues.GeoShapeValue geoShapeValue = GeoTestUtils.geoShapeValue(geometry);
+        GeometryDocValueReader reader = GeoTestUtils.geometryDocValueReader(geometry, CoordinateEncoder.GEO);
+        for (int i = 0; i < 1000; i++) {
+            LatLonGeometry latLonGeometry = supplier.get();
+            GeoRelation relation = geoShapeValue.relate(latLonGeometry);
```

Comment:
It took me a moment to realize that you are testing the new relates visitor against the existing specific visitors. Seems like a valid approach to testing this.